src/txn/txn_rollback_to_stable.c

```diff
@@ -183,22 +305,14 @@ __txn_abort_row_ondisk_kv(
         /* Stable version according to the timestamp. */
         return (0);
 
-    /* If we don't yet have a modify structure, we'll need one. */
-    WT_ERR(__wt_page_modify_init(session, page));
-    mod = page->modify;
-
-    /* Allocate an update array as necessary. */
-    WT_PAGE_ALLOC_AND_SWAP(session, page, mod->mod_row_update, upd_entry, page->entries);
-
-    /* Set the WT_UPDATE array reference. */
-    upd_entry = &mod->mod_row_update[WT_ROW_SLOT(page, rip)];
-    upd->next = NULL;
-    WT_ERR(__wt_update_serial(session, page, upd_entry, &upd, size, false));
+    WT_ERR(__txn_abort_row_replace_upd_list(session, page, rip, upd));
     WT_STAT_CONN_INCR(session, txn_rollback_upd_aborted);
     return (0);
 
 err:
-    __wt_free(session, upd);
+    if (upd != NULL)
```

Comment:
This change is not needed. `__wt_free` already handles the case if the arg is NULL.

Reply:
Fixed.